File: app/src/main/java/pozzo/apps/travelweather/model/Address.java
package pozzo.apps.travelweather.model;

import android.provider.BaseColumns;

import com.activeandroid.annotation.Column;
import com.activeandroid.annotation.Table;

/**
 * Represent a single point in map.
 *
 * Created by soldier on 10/4/15.
 */
@Table(name = "address", id = BaseColumns._ID)
public class Address extends BaseModel {
    public interface Col {
        String ADDRESS = "address";
        String LATITUDE = "latitude";
        String LONGITUDE = "longitude";
    }

    @Column(name = Col.ADDRESS)
    private String address;

    @Column(name = Col.LATITUDE)
    private double latitude;

    @Column(name = Col.LONGITUDE)
    private double longitude;

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public double getLatitude() {
        return latitude;
    }

    public void setLatitude(double latitude) {
        this.latitude = latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public void setLongitude(double longitude) {
        this.longitude = longitude;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Address address1 = (Address) o;

        if (Double.compare(address1.latitude, latitude) != 0) return false;
        if (Double.compare(address1.longitude, longitude) != 0) return false;
        return !(address != null ? !address.equals(address1.address) : address1.address != null);

    }

    @Override
    public int hashCode() {
        int result;
        long temp;
        result = address != null ? address.hashCode() : 0;
        temp = Double.doubleToLongBits(latitude);
        result = 31 * result + (int) (temp ^ (temp >>> 32));
        temp = Double.doubleToLongBits(longitude);
        result = 31 * result + (int) (temp ^ (temp >>> 32));
        return result;
    }
}

File: app/src/main/java/pozzo/apps/travelweather/map/model/Address.kt
package pozzo.apps.travelweather.map.model

/**
 * Represent a single point in map.
 */
class Address {
    var address: String? = null
    var latitude: Double = 0.0
    var longitude: Double = 0.0

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false

        val address = other as Address

        if (java.lang.Double.compare(address.latitude, latitude) != 0) return false
        return if (java.lang.Double.compare(address.longitude, longitude) != 0) false
        else !if (this.address != null) this.address != address.address else address.address != null
    }

    override fun hashCode(): Int {
        var result: Int = if (address != null) address!!.hashCode() else 0
        var temp: Long = java.lang.Double.doubleToLongBits(latitude)
        result = 31 * result + (temp xor temp.ushr(32)).toInt()
        temp = java.lang.Double.doubleToLongBits(longitude)
        result = 31 * result + (temp xor temp.ushr(32)).toInt()
        return result
    }
}
